src/controllers/productController.js

```diff
@@ -22,9 +22,48 @@ const create = async (req, res) => {
   res.redirect("/api/v1/products");
 };
 
-const showNewForm = (req, res) => {
-  res.render("products/showNewForm");
+const edit = async (req, res) => {
+  const productId = req.params.id;
+  productService
+    .findById(productId)
+    .then((prod) => {
+      prod.name = req.body.name;
+      prod.desc = req.body.desc;
+      if (req.file) {
+        // user has uploaded a new picture for the product
+        prod.imageUrl = req.file.path;
+      }
+      prod.price = req.body.price;
+      prod.quantity = req.body.quantity;
+      prod.rating = req.body.rating;
+
+      return prod.save();
+    })
+    .then(() => res.redirect(`/api/v1/products/${productId}`))
+    .catch((err) => console.error(err));
+};
+
+const showAddProductForm = (req, res) => {
+  res.render("products/productForm", { title: "Add Product", editing: false });
 };
+
+const showEditProductForm = (req, res) => {
+  const productId = req.params.id;
+  productService
+    .findById(productId)
+    .then((prod) => {
```

Comment:
Same as above, use async await and remove console instead use winston logger this applies to other instances of console.log

Reply:
done